spec/requests/timezones_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+RSpec.describe "Timezone Set", type: :request do
+  describe "GET / and /admin" do
+    before { cookies["browser.timezone"] = "Asia/Kolkata" }
+
+    it "should correctly set user timezone" do
+      get root_url
+      assert_equal 200, status
+      expect(response.body).to include(cookies["browser.timezone"])
```

Comment:
It looks like these tests are really just testing whether cookies work. Is there some element of the UI that reflects the timezone changes that we can test for?